Description: Add macros for wrapping EFI calls on X64
 Turns out uefi_call_wrapper() needs all the arguments to be UINT64. As
 uefi_call_wrapper() is a function with a variable argument list and not a
 macro, we need an additional layer of macros as we can't cast all arguments
 to UINT64 on IA32...
 .
 This fixes rEFIt on X64; symptoms were missing images and error messages
 popping up randomly, requiring a key press or a reboot.
Author: Julien BLACHE <jblache@debian.org>

Index: refit-0.14/include/refit_call_wrapper.h
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ refit-0.14/include/refit_call_wrapper.h	2010-04-21 17:25:59.526288272 +0200
@@ -0,0 +1,37 @@
+#ifndef __REFIT_CALL_WRAPPER_H__
+#define __REFIT_CALL_WRAPPER_H__
+
+#ifdef EFIX64
+# define refit_call1_wrapper(f, a1) \
+  uefi_call_wrapper(f, 1, (UINT64)(a1))
+# define refit_call2_wrapper(f, a1, a2) \
+  uefi_call_wrapper(f, 2, (UINT64)(a1), (UINT64)(a2))
+# define refit_call3_wrapper(f, a1, a2, a3) \
+  uefi_call_wrapper(f, 3, (UINT64)(a1), (UINT64)(a2), (UINT64)(a3))
+# define refit_call4_wrapper(f, a1, a2, a3, a4) \
+  uefi_call_wrapper(f, 4, (UINT64)(a1), (UINT64)(a2), (UINT64)(a3), (UINT64)(a4))
+# define refit_call5_wrapper(f, a1, a2, a3, a4, a5) \
+  uefi_call_wrapper(f, 5, (UINT64)(a1), (UINT64)(a2), (UINT64)(a3), (UINT64)(a4), (UINT64)(a5))
+# define refit_call6_wrapper(f, a1, a2, a3, a4, a5, a6) \
+  uefi_call_wrapper(f, 6, (UINT64)(a1), (UINT64)(a2), (UINT64)(a3), (UINT64)(a4), (UINT64)(a5), (UINT64)(a6))
+# define refit_call10_wrapper(f, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) \
+  uefi_call_wrapper(f, 10, (UINT64)(a1), (UINT64)(a2), (UINT64)(a3), (UINT64)(a4), (UINT64)(a5), (UINT64)(a6), (UINT64)(a7), (UINT64)(a8), (UINT64)(a9), (UINT64)(a10))
+#else
+# define refit_call1_wrapper(f, a1) \
+  uefi_call_wrapper(f, 1, a1)
+# define refit_call2_wrapper(f, a1, a2) \
+  uefi_call_wrapper(f, 2, a1, a2)
+# define refit_call3_wrapper(f, a1, a2, a3) \
+  uefi_call_wrapper(f, 3, a1, a2, a3)
+# define refit_call4_wrapper(f, a1, a2, a3, a4) \
+  uefi_call_wrapper(f, 4, a1, a2, a3, a4)
+# define refit_call5_wrapper(f, a1, a2, a3, a4, a5) \
+  uefi_call_wrapper(f, 5, a1, a2, a3, a4, a5)
+# define refit_call6_wrapper(f, a1, a2, a3, a4, a5, a6) \
+  uefi_call_wrapper(f, 6, a1, a2, a3, a4, a5, a6)
+# define refit_call10_wrapper(f, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) \
+  uefi_call_wrapper(f, 10, a1, a2, a3, a4, a5, a6, a7, a8, a9, a10)
+#endif
+
+#endif /* !__REFIT_CALL_WRAPPER_H__ */
+
